src/IO/S3/Client.cpp

```diff
@@ -344,8 +346,32 @@ Model::CreateMultipartUploadOutcome Client::CreateMultipartUpload(const CreateMu
 
 Model::CompleteMultipartUploadOutcome Client::CompleteMultipartUpload(const CompleteMultipartUploadRequest & request) const
 {
-    return doRequest(
+    auto outcome = doRequest(
         request, [this](const Model::CompleteMultipartUploadRequest & req) { return Aws::S3::S3Client::CompleteMultipartUpload(req); });
+
+    if (provider_type != ProviderType::GCS)
+        return outcome;
+
```

Comment:
Do we need to check `outcome.IsSuccess()` here and not trying to do Compose after failure?